packages/keyring-controller/src/KeyringController.ts

```diff
@@ -908,6 +945,24 @@ export class KeyringController extends BaseController<
     return keyring.decryptMessage(address, messageParams.data);
   }
 
+  /**
+   * Returns the keyring with the given id.
+   *
+   * @param id - The id of the keyring to return.
+   * @returns The keyring with the given id.
+   */
+  getKeyringById(id: string): unknown {
+    const keyring = this.#keyrings.find(
+      (item) =>
+        (item as EthKeyring<Json> & { opts: { id: string } }).opts.id === id,
+    );
+    if (!keyring) {
+      throw new Error(KeyringControllerError.KeyringNotFound);
+    }
+
+    return keyring;
+  }
+
```

Comment:
This method is unsafe as clients (consumers) should not be able to access a keyring instance directly. That's the reason why `getKeyringsBytype` and `getKeyringForAccount` have been deprecated.

One suggestion could be to make this a `#` method so that consumers are forced to pass through `withKeyring`